axi_intercon_gen.py

```diff
@@ -0,0 +1,552 @@
+        ("bresp" , True , 2),
+        ("bvalid", True , 0),
+        ("buser"  , True, w.user),
+        ("bready", False, 0),
+
+        ("rid"   , True , id_width),
+        ("rdata" , True , w.data),
+        ("rresp" , True , 2),
+        ("rlast" , True , 0),
+        ("ruser"  , True, w.user),
+        ("rvalid", True , 0),
+        ("rready", False, 0),
+    ]
+    if drop_prefix:
+        signals2 = []
+        for s in signals:
+            sname = s[0]
+            if sname.startswith("aw"):
+                sname = sname[2:]
+            signals2.append((sname, s[1], s[2]))
+        signals = signals2
+    return signals
+
+def module_ports(w, intf, id_width, is_input):
+    ports = []
+    for s in axi_signals(w, id_width):
+        if s[0].endswith('user') and not w.user:
+            continue
+        if s[0].startswith('aw') and intf.read_only:
+            continue
+        if s[0].startswith('w') and intf.read_only:
+            continue
+        if s[0].startswith('b') and intf.read_only:
+            continue
+        prefix = 'o' if is_input == s[1] else 'i'
+        ports.append(ModulePort("{}_{}_{}".format(prefix, intf.name, s[0]),
+                                'output' if is_input == s[1] else 'input',
+                                s[2]))
+    return ports
+
+def assigns(w, max_idw, masters, slaves):
+    raw = '\n'
+    i = 0
+    for m in masters:
+        raw += "   //Master {}\n".format(m.name)
+        for (ch, name, _dir, width) in axi_signals2(w, m.idw):
+            if name in ['valid','ready']:
+                _name = ch+'_'+name
+            else:
+                _name = ch+'.'+name
+            if name == 'user' and not w.user:
+                continue
+            if _dir:
+                if m.read_only and ((ch == 'aw') or (ch == 'w') or (ch == 'b')):
+                    continue
+
+                src = "masters_resp[{}].{}".format(i, _name)
+                if ch in ['b', 'r'] and name == 'id' and m.idw < max_idw:
+                    src = src+'[{}:0]'.format(m.idw-1)
+                raw += "   assign o_{}_{}{} = {};\n".format(m.name, ch, name, src)
+            else:
+                if name == 'atop':
+                    if ch == 'aw':
+                        src = "6'd0"
+                    else:
+                        continue
+                src = "i_{}_{}{}".format(m.name, ch, name)
+                if ch in ['ar', 'aw'] and name == 'id' and m.idw < max_idw:
+                    src = "{"+ str(max_idw-m.idw)+"'d0,"+src+"}"
+                if m.read_only and (ch in ['aw' , 'w', 'b']):
+                    if ch == 'aw' and name == 'id':
+                        _w = max_idw
+                    else:
+                        _w = max(1,width)
+                    src = "{}'d0".format(_w)
+                raw += "   assign masters_req[{}].{} = {};\n".format(i, _name, src)
+#        raw += "   assign connectivity_map[{}] = {}'b{};\n".format(i, len(slaves), '1'*len(slaves))
```

Comment:
Can code that is commented out be removed?

Reply:
Ouch. I see now that the file is a bit messy. Will sleep before pushing next time :)